redis/src/aio/mod.rs

```diff
@@ -140,6 +140,20 @@ where
         }
     }
 
+    if connection_info.client_name.is_some() {
+        match cmd("CLIENT")
+            .arg("SETNAME")
+            .arg(connection_info.client_name.as_ref().unwrap())
+            .query_async(con).await
+        {
+            Ok(Value::Okay) => {}
+            _ => fail!((
```

Comment:
since the returned value of `query_async` is `RedisError`, you don't need to fail, just use the `?` operator.

Reply:
Again this is the same as SELECT command just above

Reply:
beyond this PR - will refactor in different PR